libraries/AP_Avoidance/AvoidanceHandler.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <AP_Avoidance/AP_Avoidance.h>
+
+class AP_Avoidance;
+
+class AvoidanceHandler {
+public:
+
+    // enter is called once when starting this avoidance action
+    virtual bool enter(class AP_Avoidance::Sample &threat, class AvoidanceHandler *old_handler);
```

Comment:
I don't think so; several of the current avoidance handlers don't need to do any work not one by the base class code.